src/ui/views/TradeFilter/FilterGroup.tsx

```diff
@@ -0,0 +1,74 @@
+import React, { useEffect, useState } from "react";
+import { ToggleItem } from "../../components/ToggleItem";
+import type { filterType } from "./TradeFilter";
+
+export default function FilterGroup(props: {
+	title: string;
+	statuses: filterType;
+	values: string[];
+	type: string;
+	onUpdate: (type: string, value: string[]) => void;
+}) {
+	const [activeFilters, setActiveFilters] = useState<string[]>(
+		props.statuses[props.type].filterData,
+	);
```

Comment:
I think it would be better to make this a Set rather than an Array. Then your code below to add/remove entries is just `activeFilters.add(value)` or `activeFilters.remove(value)` rather than having to worry about indexOf, splice, and destructuring.

That would break your `useEffect` that watches `activeFilters` though... but I think that would be more clearly accomplished by making a wrapper function `setActiveFiltersAndUpdate` that calls `setActiveFilters` and then calls `props.onUpdate`. IMHO it gets a little confusing when `useEffect` is relied on for some logic that could just linearly be implemented in a function.